### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by Julia-ivv**
1. Service is done, for example
  https://l6mg2h5d9g.execute-api.eu-west-1.amazonaws.com/dev/products
  https://l6mg2h5d9g.execute-api.eu-west-1.amazonaws.com/dev/products/0
   FE works.
   https://d2e65jaievfvyp.cloudfront.net/
2. Link to product-service API - https://l6mg2h5d9g.execute-api.eu-west-1.amazonaws.com/dev/
3. LInk to FE MR - https://github.com/Julia-ivv/nodejs-aws-fe/pull/2
4. Async/await is used in lambda functions
5. ES6 modules are used 
6. Webpack is configured for product-service
7. Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.
8. Main error scenarious are handled by API ("Product not found" error, try catch blocks are used in lambda handlers).

### CODE REVIEW

File: product-service/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  /products/{productId}:
+    get:
+      summary: Returns a product by Index.
+      description: Returns a product by Index.
+      responses:
+        '200':
+          description: A JSON object of a product
+          content:
+            application/json:
+              schema:
```
Comment (IvanPresmytsky) [inline product-service/swagger.yaml:34]: The descriptive schema for the Product should be provided.
```
{
  id: nmber,
  title: string;
  and ect.
}
```

File: product-service/handlers/getProductsList.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  try {
+    return {
+        statusCode: 200,
+        headers: {
+            // "Access-Control-Allow-Credentials" : true // Required for cookies, authorization headers with HTTPS
+            'Content-Type': 'application/json',
+            'Access-Control-Allow-Origin': '*',
+            'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept',
+            'Access-Control-Allow-Methods': 'POST, GET, OPTIONS, PATCH, PUT',
+        },
```
Comment (IvanPresmytsky) [inline product-service/handlers/getProductsList.js]: It worth moving these headers to the const and use between handlers.

File: product-service/handler.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { getProductsList } from './handlers/getProductsList.js';
```
Comment (IvanPresmytsky) [inline product-service/handler.js:1]: you can export them like that:
`export * from './handlers/getProductsList.js';`

**Review by IvanPresmytsky — APPROVED** [review]

File: product-service/swagger.yaml (hunk 1 above)
Comment (Julia-ivv) [reply product-service/swagger.yaml:34]: fixed it

File: product-service/handlers/getProductsList.js (hunk 2 above)
Comment (Julia-ivv) [reply product-service/handlers/getProductsList.js]: changed it

File: product-service/handlers/getProductsById.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import productList from './productList.json';
+import { headers } from '../constants/headers';
+
+export const getProductsById = async event => {
+  const { productId } = event.pathParameters;
+  try {
```
Comment (Emerkof) [inline product-service/handlers/getProductsById.js:7]: I'd suggest moving duplicate logic (like error handling, response formatting) into separate `lib/` functions and use them across the handlers.

It's a wish for the future, no need to implement it right now. It won't affect your score.

File: product-service/handlers/getProductsById.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    if (productId > productList.length - 1) throw new Error();
+
+    return {
+        statusCode: 200,
+        headers: headers,
+        body: JSON.stringify(productList[productId]),
```
Comment (Emerkof) [inline product-service/handlers/getProductsById.js]: I found it confusing, that I can't search products by their actual Ids, like `7567ec4b-b10c-45c5-9345-fc73c48a80a1`. Could you please fix it to something like `productList.find(product => product.id === id)`. And throw error, if a product is not found.

**Comment by Emerkof** [comment]
@Julia-ivv 

> Async/await is used in lambda functions

Can't find any, could you please add `async/await` usage?

File: product-service/handlers/getProductsById.js (hunk 4 above)
Comment (Julia-ivv) [reply product-service/handlers/getProductsById.js:7]: changed it

File: product-service/handlers/getProductsById.js (hunk 5 above)
Comment (Julia-ivv) [reply product-service/handlers/getProductsById.js]: changed it

**Comment by Julia-ivv** [comment]
> @Julia-ivv
> 
> > Async/await is used in lambda functions
> 
> Can't find any, could you please add `async/await` usage?

**Comment by Julia-ivv** [comment]
PR was closed by mistake

File: product-service/handlers/getProductsById.js (hunk 5 above)
Comment (Julia-ivv) [reply product-service/handlers/getProductsById.js]: and added async/await usage

Comment (Emerkof) [reply product-service/handlers/getProductsById.js]: acknowledged + approved

**Review by Shestak2039 — APPROVED** [review]
Well done!

**Pull request merged**